### PR TITLE
Module level reordering

### PR DESCRIPTION
**Pull request by shuheiktgw**
### Overview

nose-launchable eagerly loads tests when preparing a test in order to retrieve test cases from it. However, that behavior conflicts with test generators since nose-launchable needs to execute the generator to retrieve test cases from it. To avoid that, this PR fixed the plugin to stop loading tests once it encounters a module (or a class in [a special case](https://github.com/launchableinc/nose-launchable/pull/4/files#diff-0e0177f71aa907fd90a598f4a6ea794f682a849263e079e1390184f4fb1cdcb1R95)) and reorders tests at a module level.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -29,4 +29,4 @@ jobs:
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
     - name: Test with pytest
       run: |
-        pytest launchable/tests
```
Comment (shuheiktgw) [inline .github/workflows/test.yaml:32]: Moved test directory from launchable/tests/ to tests/

File: launchable/manager.py [hunk 2]
```diff
@@ -68,6 +76,37 @@ def _reorder(test, order):
...
             tree_nodes.append((node, child))
 
-    test._tests = tests
+    suite._tests = tests
 
     return tree_nodes
+
+
+# Check if test's context is a leaf
+def _is_leaf(suite):
```
Comment (shuheiktgw) [inline launchable/manager.py:85]: This is the single most important function I added in this PR. There are three possible cases we need to stop loading as I described below comment:

1. When the context is Failure, that means nose fails to load the specified test file the node will be a leaf and we don't need to go down further.
2. If the context is Module, there are two possibilities 1. the module is a package 2. the module is a file. If the module is a package i.e. `__file__` ends with `__init__.py` we need to go down further and look for a file.
3. If the context is a class we should stop loading there. This is a relatively rare case when a user specifies a class name directly

File: launchable/manager.py [hunk 3]
```diff
@@ -68,6 +76,37 @@ def _reorder(test, order):
...
+        # a leaf context can be a class if specifying a class directly: module:TestClass
+        is_class = type(context) is type
+
+        return is_failure or is_file or is_class
+
+    return False
+
+
+# Extract a test name from a leaf
+def _get_test_name(suite):
```
Comment (shuheiktgw) [inline launchable/manager.py:104]: We use a module name as a `testName`

File: launchable/plugin.py [hunk 4]
```diff
@@ -44,8 +44,8 @@ def prepareTest(self, test):
 
             reorder(test, order)
 
-            self._print("Test execution optimized by Launchable 🚀")
-            return
+            self._print("Test execution optimized by Launchable")
```
Comment (ninjinkun) [inline launchable/plugin.py:47]: Why did you remove 🚀 ? 🤔

File: launchable/manager.py (hunk 2 above)
Comment (ninjinkun) [reply launchable/manager.py:85]: This function is very clear. I can read your purpose easily from this code. I love this function ❤️

File: .gitignore [hunk 5]
```diff
@@ -3,4 +3,4 @@
 dist/
 build/
 launchable/__pycache__
-launchable/tests/__pycache__
+tests/__pycache__
```
Comment (ninjinkun) [inline .gitignore:6]: Add `__pycache__/` for other directories. Or https://github.com/github/gitignore/blob/master/Python.gitignore is much better

**Review by ninjinkun — APPROVED** [review]
I approved the basic design

File: launchable/plugin.py (hunk 4 above)
Comment (shuheiktgw) [reply launchable/plugin.py:47]: Oh, sorry I was intended to leave a comment here and totally forgot about it!
I was reported that the emoji does not work in our customer's environment but I think we can try using Unicode instead. Let me try!

File: .gitignore (hunk 5 above)
Comment (shuheiktgw) [reply .gitignore]: That looks way much better! Thank you!

**Comment by shuheiktgw** [comment]
@ninjinkun Thank you for your review! I'll merge this PR.

**Comment by shuheiktgw** [comment]
This PR conflicts with other PRs so let me port those changes to those PRs later
